### ISSUE TITLE
Error compiling on armhf platform

### ISSUE THREAD

**Issue by CodingSinceThe80s** [miracl/amcl#56]
Hi everyone,

I am getting an error when compiling the library as part of hyperledger fabric (fabric-amcl) on a raspberry pi 4 (raspbian, armhf architecture). As far as I can tell it is due to the int representation on 32-bit platforms which could be easily fixed. Here is the error:

```
Building build/bin/orderer
GOBIN=/go/src/github.com/hyperledger/fabric/build/bin go install -tags "" -ldflags "-X github.com/hyperledger/fabric/common/metadata.Version=2.2.1 -X github.com/hyperledger/fabric/common/metadata.CommitSHA=344fda6 -X github.com/hyperledger/fabric/common/metadata.BaseDockerLabel=org.hyperledger.fabric -X github.com/hyperledger/fabric/common/metadata.DockerNamespace=hyperledger" github.com/hyperledger/fabric/cmd/orderer
# github.com/hyperledger/fabric-amcl/amcl/FP256BN
vendor/github.com/hyperledger/fabric-amcl/amcl/FP256BN/FP.go:110:18: constant 30564559323915749 overflows int
vendor/github.com/hyperledger/fabric-amcl/amcl/FP256BN/FP.go:394:11: constant 30564559323915749 overflows int
vendor/github.com/hyperledger/fabric-amcl/amcl/FP256BN/FP.go:394:20: constant 30564559323915754 overflows int
vendor/github.com/hyperledger/fabric-amcl/amcl/FP256BN/FP.go:394:25: constant 3820569915489469 overflows int
vendor/github.com/hyperledger/fabric-amcl/amcl/FP256BN/FP.go:397:11: constant 30564559323915749 overflows int
vendor/github.com/hyperledger/fabric-amcl/amcl/FP256BN/FP.go:397:20: constant 30564559323915752 overflows int
vendor/github.com/hyperledger/fabric-amcl/amcl/FP256BN/FP.go:397:25: constant 7641139830978938 overflows int
make: *** [Makefile:209: build/bin/orderer] Error 2

```
If no fix is desired please also let me know, eventually what I could do as a workaround. Or am I the only crazy person trying to compile the hyperledger fabric images on 32-bit architecture? :) I am trying to do this in order to lower the hurdles in deploying hyperledger fabric on standard raspberry pis for people who are not well versed with blockchain and/or uncommon cpu architectures.

**Comment by mcarrickscott** [comment]
Hello,

The AMCL library supports both 32 and 64-bit versions. Here it would appear
that a 64-bit build is being used on a 32-bit architecture.

So solution is to use a 32-bit build of the library. See
https://github.com/miracl/amcl/blob/master/version3/go/readme.txt

Mike

**Comment by CodingSinceThe80s** [comment]
Hello Mike, 

thanks for the tip, I will try this and post the result here.

Best regards,
Razvan

**Comment by CodingSinceThe80s** [comment]
Hi again, I have tried the steps and the TL;DR; of it is: it worked :) so many thanks with that!

The long version of it is that I had to do a bit more work to integrate the 32-bit-version back into the fabric-amcl repository:

1. Used the python-script to change to 32-bit as in the amcl-readme file
2. Selected only FP265BN (the only one in fabric-amcl) when prompted by the script (option 22 I believe)
3. Copied the generated amcl folder over the old repo (locally inside the vendor folder structure, I did not push the changes)
4. Changed the import directives for some of the files so that the do not point to the miracl-repository but rather to the fabric-amcl repo, so that the import looked like this: `import "github.com/hyperledger/fabric-amcl/amcl"`

Build process for hyperledger fabric then worked without any errors.
